File: sample/src/androidTest/java/com/github/grigory_rylov/instrumentaltestwithannotations/SampleAnnotation.java
package com.github.grigory_rylov.instrumentaltestwithannotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface SampleAnnotation {
    String stringParam();

    int intParam() default 0;

    int[] intArray() default {};

    String[] strArray() default {};
}

File: sample/src/androidTest/java/com/github/grigory_rylov/instrumentaltestwithannotations/SampleAnnotation.kt
package com.github.grigory_rylov.instrumentaltestwithannotations

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.CLASS,
        AnnotationTarget.FILE,
        AnnotationTarget.FUNCTION,
        AnnotationTarget.PROPERTY_GETTER,
        AnnotationTarget.PROPERTY_SETTER)
annotation class SampleAnnotation(
        val stringParam: String,
        val intParam: Int,
        val intArray: IntArray,
        val strArray: Array<String>
)
